crates/red_knot_python_semantic/src/types.rs

```diff
@@ -4579,7 +4579,7 @@ impl<'db> GeneralCallableType<'db> {
         if signature.parameters().iter().any(|parameter| {
             parameter
                 .annotated_type()
-                .is_some_and(|annotated_type| !annotated_type.is_fully_static(db))
+                .is_none_or(|annotated_type| !annotated_type.is_fully_static(db))
         }) {
             return false;
         }
```

Comment:
I think we need to make the same change for the `signature.return_ty` code immediately below this?

Reply:
I don't think so because `is_some_and` would return false if the `return_ty` is `None` which would make it not fully static. It's only in this case because we short-circuit using `any`.

Reply:
Sorry, you're quite right!